### DISCUSSION TITLE
Conflict between the libcrypto.so and libssl.so native libraries when using both srt and rtmp dependencies.

### DISCUSSION THREAD

**Discussion by cristiAndreiTarasi** [ThibaultBee/StreamPack#169] [General]
The solution provided in the demo-camera for solving the conflict between libcrypto.so and libssl.so does not work when we use the extensions as implementation dependencies, ex: 

```
dependencies {
    implementation("io.github.thibaultbee:streampack:2.6.0")
     implementation ("io.github.thibaultbee:streampack-extension-rtmp:2.6.0")
    implementation("io.github.thibaultbee:streampack-extension-srt:2.6.0")
}
```
Error `2 files found with path 'lib/arm64-v8a/libcrypto.so' from inputs:` occurs.  So the solution from README doesn't work:
```
 android {
    packaging {
        jniLibs {
            pickFirsts += "**/*.so"
        }
    }
}
```

I am only able to use only rtmp or srt, and I couldn't find a way to fix this so far.

**Comment by ThibaultBee** [comment]
Hi,

The demos from the project uses this solution. Have a look at https://github.com/ThibaultBee/StreamPack/blob/ab8027231982e5e8280fb99812dcd6d0ef28440d/demos/camera/build.gradle#L45

Did you add it in a library or application build.gradle? What AGP version?

**Comment by cristiAndreiTarasi** [comment]
I used it in a library build.gradle and AGP is 8.2.2. I managed eventually to make it work in the following way:

   ```
 implementation("io.github.thibaultbee:streampack:2.6.0")
    
    // Include SRT extension and allow it to package libcrypto and libssl
    implementation("io.github.thibaultbee:streampack-extension-srt:2.6.0")

    // Include RTMP extension but exclude its libcrypto and libssl libraries
    implementation("io.github.thibaultbee:streampack-extension-rtmp:2.6.0") {
        exclude(group = "video.api", module = "rtmpdroid")
    }
    
    // Re-add rtmpdroid excluding the problematic libraries
    implementation("video.api:rtmpdroid:1.2.1-packed") {
        exclude(group = "video.api", module = "libcrypto.so")
        exclude(group = "video.api", module = "libssl.so")
    }
```

[api.video-rtmpdroid](https://github.com/apivideo/api.video-rtmpdroid) has a `-packed` option.

**Reply by ThibaultBee** [reply]
The drawback of doing that is that the resulted APK size is bigger that with `packagingOptions`.

I guess it only works in an application build.gradle.

```
// Re-add rtmpdroid excluding the problematic libraries
 implementation("video.api:rtmpdroid:1.2.1-packed") {
     exclude(group = "video.api", module = "libcrypto.so") // useless
     exclude(group = "video.api", module = "libssl.so") // useless
 }
```

**Comment by ThibaultBee** [comment]
Should not happen with 3.X anymore because srtdroid library embed these libraries as a fat lib.

Moreover, I plan to change the RTMP provider in 3.1 to a Kotlin solution.
